src/ImageSharp.Web/DependencyInjection/ImageSharpBuilderExtensions.cs

```diff
@@ -377,7 +371,28 @@ public static IImageSharpBuilder Configure<TOptions>(this IImageSharpBuilder bui
              where TOptions : class
         {
             builder.Services.Configure(configureOptions);
+
+            return builder;
+        }
+
+        /// <summary>
+        /// Registers an action used to configure a particular type of options. Note: These are run after all <see cref="Configure{TOptions}(IImageSharpBuilder, Action{TOptions})"/>.
+        /// </summary>
+        /// <typeparam name="TOptions">The options type to be configured.</typeparam>
+        /// <param name="builder">The core builder.</param>
+        /// <param name="configureOptions">The action used to configure the options.</param>
+        /// <returns>The <see cref="IImageSharpBuilder"/>.</returns>
+        public static IImageSharpBuilder PostConfigure<TOptions>(this IImageSharpBuilder builder, Action<TOptions> configureOptions)
```

Comment:
What benefit do you see to this rather than having the explicit `Configure<T>` methods for each type? 

`PostConfigure<T>` is designed to be called from within extension methods to add services to ensure that those options are not overwritten. It doesn't really have a place in the fluent builder API.

Reply:
I wasn't sure why `Configure<T>()` was added to `IImageSharpBuilder` at first, as it's just a proxy to `IServiceCollection`, but it makes it a lot easier to configure ImageSharp related options in a fluent way (although you can configure any options).

Maybe it's indeed a sensible idea to add overloads for the specific ImageSharp option types, so it easier to discover relevant options, although that would mean the Azure and S3 packages will also need to provide their own overloads for their types. You could also introduce an decorator interface and limit the generic type to e.g. `where TOptions : IImageSharpOptions`, but that would make discovery of the available ImageSharp options a bit harder again. What's your opinion on this?

As for `PostConfigure<T>()`, that basically does the same as `Configure<T>()`, but is invoked last, ensuring you have the final state of the options (in e.g. case multiple configure calls are done).

Reply:
Had a quick look and you can't create overloads for `Configure<T>(this IImageSharpBuilder builder, IConfiguration config)` that specify specific ImageSharp options, so the only way to limit the configure methods on `IImageSharpBuilder` is to create a marker interface...

I think only allowing ImageSharp related configuration on the `IImageSharpBuilder` is actually a better API design and you can still (post)configure the options directly on the `IServiceCollection` if you want...

Reply:
Marker interfaces are a code smell.

Configure is fine as it is, the flexibility is a strength and it works really well when building the configuration sequentially. The examples in the Samples startup are indicative of that strength as does the config in the the unit tests.